subsys/net/lib/lwm2m/lwm2m_rd_client.c

```diff
@@ -1521,5 +1590,22 @@ static int lwm2m_rd_client_init(const struct device *dev)
 
 }
 
-SYS_INIT(lwm2m_rd_client_init, APPLICATION,
-	 CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
+#ifdef CONFIG_LWM2M_BLOCK_TRANSFER_TESTS
+/* These wrappers are used for tests to get access to
+ * static functions. They need to be declared manually in the
+ * tests before use. */
+void test_lwm2m_rd_client_setup(struct lwm2m_ctx *ctx)
```

Comment:
I'm really not a fan of introducing test code directly into the implementation. If you need to access static variables/fucntions, just include the `*.c` file in the test file.